src/components/icon/icon.tsx

```diff
@@ -0,0 +1,63 @@
+import { Component, Host, State, Prop, h, Watch } from '@stencil/core';
+import { getSvgPath, getSvgContent, formatSvg } from './utils';
+
+@Component({
+  tag: 'bds-icon',
+  styleUrl: 'icon.scss',
+  assetsDir: 'svg',
+  shadow: true
+})
+export class Icon {
+  @State() svgContent?: string;
+
+  /**
+  * Specifies which icon to use from the built-in set of icons.
+  */
+  @Prop({ reflect: true }) name!: string;
+
+  /**
+  * Specifies the label to use for accessibility. Defaults to the icon name.
+  */
+  @Prop({ mutable: true, reflectToAttr: true }) ariaLabel?: string;
+
+  /**
+   * Icon size. Entered as one of the icon size design tokens. Can be one of: "xxx-small", "xx-small", "x-small", "small", "medium", "large", "x-large", "xx-large", "xxx-large".
+   */
+  @Prop() size?: string = 'medium';
+
+  /**
+  * Specifies the color to use.Specifies a color to use. The default is svg.
+  */
+  @Prop() color?: string;
+
+  /**
+  * Specifies the theme to use outline or solid icons. Defaults to outline.
+  */
+  @Prop({ reflect: true }) theme: 'outline' | 'solid' = 'outline';
+
+  async connectedCallback(): Promise<void> {
+    await this.loadSvg();
+  }
+
+  @Watch('name')
```

Comment:
[ts-nameof package](https://www.npmjs.com/package/ts-nameof) makes it easy to refactor class properties if it necessary in the future. I added it on blip-portal and I think that we can discuss about it uses right on the start of design system